### ISSUE TITLE
Should  `transformer-directives` remove original atomic classes from the result?

### ISSUE THREAD

**Issue by dongivan** [unocss/unocss#1267]
`transformer-directives` will transform `.temp { @apply w-12; }` to `.temp { width: 3rem; }`, and then the generated `.temp` and atomic class `.w-12` will all be output to the built css file.
Should only `.temp` be output in this situation?

I've made a demo: (https://stackblitz.com/edit/vitejs-vite-eqcbfz?file=dist%2Fassets%2Findex.09493d79.css)
In the `dist/assets/index.09493d79.css` (which is generated by `vite build` command), `.rectangle` and its atomic classes `.w-12`, `.h-12` and `.bg-red-400` are all output, however `transformer-compile-class` transforms `:uno: w-6 h-6 bg-black` into `.uno-qyevtb` and only the latter is output.

**Comment by antfu** [comment]
This is actually a Vue problem. Internally in Vite's pipeline, Vue SFC will be compiled into 3 virtual modules (`template`, `script`, `style`, etc.), where the original SFC also goes to the pipeline. So `transformer-directives` actually transformed the virtual CSS but not the SFC (at that time the plugin did not understand what is CSS in this SFC) and caused the main scanned utils in the SFC.

I don't expect Vue will change this approach soon, so as a workaround, you can move the style tag to a real `.css` file and it should work.

**Comment by dongivan** [comment]
Sorry for that I don't understand your explanation because of my poor English.
But your suggestion (move the style tag to a real `.css` file) works well! Thank you!
